documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -834,6 +831,103 @@ public interface IJoystickInputHandler : IInputHandler
 
 `HandleHatMove` must be called when any value of `JoystickState.Hats` changes.
 
+# Input Maps
+
+## Presentation
+Input maps are designed to be easyer to the user to handle actions with more than one input.
+At first view it can looks useless and a bad practice, but this is complemtelly useful for cross-plataform programs, being games an example.
+
+This is an pseudocode example of what the Input Map system solves:
+
+Without inputMaps:
+```cs
+inputContext.Update();
+
+private Dictionary<string, bool> pressedKeys = new();
+
+inputContext.Gamepads.ThumbstickMove += @event => {Move(@event.Value)};
+inputContext.Joysticks.AxisMove += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+With inputMaps:
+```cs
+inputContext.Update();
+
+// This single line get rid of the two in the example above. TWO ISN'T THE LIMIT!
+inputContext.Map.GetAction("Move").Axis += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+## Implementation
+Each InputContext have one reference to an InputMap.
+This is how the InputMap class work:
+```cs
+public partial class InputMap
+{
+    public InputMapAction GetAction(string name);
+    public void AddAction(InputMapAction action, string name);
+    public void RemoveAction(string name);
+    public static InputMap Load(string path);
+    public static void Save(InputMap source, string path);
+    public static string Compile(InputMap source);
+}
+```
+
+`GetAction` method is the main entry point of the system. It allows the use to get an mapped action by it configurated name.
+
+```cs
+public partial class InputMapAction
+{
+    public bool IsPressed { get; }
+    public event Action<AxisInputEvent> Axis;
+}
+```
+
+## Configuration and saving
+The input map system's save system is one of the most important to it working. It helps to make the configuration easyer as faster.
+
+Instead of needing to hardcode all the actions, the user should create a file with a special syntax. This is an example of the sintax of the map system's saved data:
+```
+"move_up":
+    - KeyName.Up
+    - JoystickButton.DPadUp
+
+"move_down":
+- KeyName.Down
+- JoystickButton.DPadDown
+
+"move":
+    - Vector2D [
+        -KeyName.Down
+        -KeyName.Up
+        -KeyName.Left
+        -KeyName.Right
+    ]
+    - Joystick.Axis
```

Comment:
Which joystick will this pick up on? Any?

Reply:
I need to say, I don't understand so much what devices "joystick" and "gamepad" are referring. The idea should be any, but now I'm thinking about the fact that a ps4 controller have 2 thumbsticks and i forgot it completely lol

Can you explain me a little what exactly these devices represent? Like i really never head something called anything for these names (obviously in some cases but pretty obvious ones like the atari controller or the one that some flying machines uses, that's i think is out of where silk acts lol), just everything as controller

Also like what input will be returned by these weird car controllers or joysticks (are these joysticks that it refers?)?